behaviour/graql/language/define.feature

```diff
@@ -1018,92 +928,695 @@ Feature: Graql Define Query
+      """
+    Given the integrity is validated
+
+    When get answers of graql query
+      """
+      match $x sub requirement; get;
+      """
+    Then concept identifiers are
+      |     | check | value            |
+      | REQ | label | requirement      |
+      | TLR | label | tool-requirement |
+    Then uniquely identify answer concepts
+      | x   |
+      | REQ |
+      | TLR |
+
+
+  Scenario: define abstract attribute type creates an abstract type
+    Given graql define
+      """
+      define number-of-limbs sub attribute, abstract, value long;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x type number-of-limbs; $x abstract; get;
+      """
+    Then concept identifiers are
+      |     | check | value           |
+      | NOL | label | number-of-limbs |
+    Then uniquely identify answer concepts
+      | x   |
+      | NOL |
+
+
+  Scenario: define concrete subtype of abstract attribute creates child of its parent type
+    Given graql define
+      """
+      define
+      number-of-limbs sub attribute, abstract, value long;
+      number-of-legs sub number-of-limbs;
+      """
+    Given the integrity is validated
+
+    When get answers of graql query
+      """
+      match $x sub number-of-limbs; get;
+      """
+    Then concept identifiers are
+      |     | check | value           |
+      | NOL | label | number-of-limbs |
+      | NLE | label | number-of-legs  |
+    Then uniquely identify answer concepts
+      | x   |
+      | NOL |
+      | NLE |
+
+
+  Scenario: define abstract subtype of abstract attribute creates child of its parent type
+    Given graql define
+      """
+      define
+      number-of-limbs sub attribute, abstract, value long;
+      number-of-artificial-limbs sub number-of-limbs, abstract;
+      """
+    Given the integrity is validated
+
+    When get answers of graql query
+      """
+      match $x sub number-of-limbs; get;
+      """
+    Then concept identifiers are
+      |     | check | value                      |
+      | NOL | label | number-of-limbs            |
+      | NAL | label | number-of-artificial-limbs |
+    Then uniquely identify answer concepts
+      | x   |
+      | NOL |
+      | NAL |
+
+
+  Scenario: define a rule as abstract throws
+    Then graql define throws
+      """
+      define
+      nickname sub name;
+      person has nickname;
+      robert-has-nickname-bob sub rule, abstract,
+      when {
+        $p isa person, has name "Robert";
+      }, then {
+        $p has nickname "Bob";
+      };
+      """
+
+
+  @ignore
+  # TODO: re-enable when concrete types are not allowed to have abstract subtypes
+  Scenario: define abstract subtype of concrete entity throws an error
+    Then graql define throws
+      """
+      define
+      exception sub entity;
+      grakn-exception sub exception, abstract;
+      """
+
+
+  Scenario: repeatedly define type as abstract creates an abstract type
```

Comment:
Should this be called `defining type as abstract is idempotent`?

Reply:
Renamed.